ui/src/router.js

```diff
@@ -243,4 +199,74 @@ router.beforeEach((to, from, next) => {
   }
 });
 
+function checkPageRole(roleName,routeName)
+{
+  var resCheck=false;
+  switch(routeName)
+  {
+    case "search-people":
+      if(searchPeopleAllowedRoles.includes(roleName))
+      {
+        store.state.allowToAccessTheNextPage = true;
+      }
+      else{
+        store.state.allowToAccessTheNextPage = false;
+      }
+      break;
+    case "add-people":
+      if(editPeopleAllowedRoles.includes(roleName))
+      {
+        store.state.allowToAccessTheNextPage = true;
+      }
+      else{
+        store.state.allowToAccessTheNextPage = false;
+      }
+      break;
+    case "edit-people":
+      if(editPeopleAllowedRoles.includes(roleName))
+      {
+        store.state.allowToAccessTheNextPage = true;
+      }
+      else{
+        store.state.allowToAccessTheNextPage = false;
+      }
+      break;
+      case "people-view":
+        if(viewPeopleAllowedRoles.includes(roleName))
+        {
+          store.state.allowToAccessTheNextPage = true;
+        }
+        else{
+          store.state.allowToAccessTheNextPage = false;
+        }
+        break;
+        case "relationship":
+          if(relationshipsAllowedRoles.includes(roleName))
+          {
+            store.state.allowToAccessTheNextPage = true;
+          }
+          else{
+            store.state.allowToAccessTheNextPage = false;
+          }
+          break;
+        case "admin-add-user":
+          if(manageUserAllowedRoles.includes(roleName))
+          {
+            store.state.allowToAccessTheNextPage = true;
+          }
+          else{
+            store.state.allowToAccessTheNextPage = false;
+          }
+          break;
+          case "admin-users":
+            if(manageUserAllowedRoles.includes(roleName))
+            {
+              store.state.allowToAccessTheNextPage = true;
+            }
+            else{
+              store.state.allowToAccessTheNextPage = false;
```

Comment:
I'm also not wild about using the store to control access. That means we'd need to insert this code on every page we create which means we will very likely miss something. Look at how we are doing the redirect if a user is not logged in:

```
      next({
        path: "/login"
      });
```

We should be doing something like that rather than checking each individual page.

Reply:
Instead of usingindividual pages to show noaccess right errors. I have created a noaccess.vue where the routes are redirected if the users do not have access